client/src/main/java/hudson/plugins/swarm/SwarmClient.java

```diff
@@ -368,6 +409,41 @@ private static String getChildElementString(Element parent, String tagName) {
         return null;
     }
 
+    /**
+     * Returns a hash that should be consistent for any individual swarm client (as long as it has a persistent IP)
+     * and should be unique to that client.
+     *
+     * @param remoteFsRoot the file system root should be part of the hash (to support multiple swarm clients from 
+     *                     the same machine)
+     * @return our best effort at a consistent hash
+     */
+    public static String hash(File remoteFsRoot) {
+        StringBuilder buf = new StringBuilder();
+        try {
+            buf.append(remoteFsRoot.getCanonicalPath()).append('\n');
+        } catch (IOException e) {
+            buf.append(remoteFsRoot.getAbsolutePath()).append('\n');
+        }
+        try {
+            for (NetworkInterface ni : Collections.list(NetworkInterface.getNetworkInterfaces())) {
+                for (InetAddress ia : Collections.list(ni.getInetAddresses())) {
+                    if (ia instanceof Inet4Address) {
+                        buf.append(ia.getHostAddress()).append('\n');
+                    } else if (ia instanceof Inet6Address) {
+                        buf.append(ia.getHostAddress()).append('\n');
```

Comment:
Not sure why you don't want to include non IP adapters (it's just more entropy?) or collapse this into on if but not enough to fail the review.